app/lib/team.js

```diff
@@ -236,17 +240,34 @@ async function removeMember (teamId, osmId) {
  */
 async function assignModerator (teamId, osmId) {
   const conn = await db()
+  if (!await isMember(teamId, osmId)) {
+    throw new Error('cannot assign osmId to be moderator because they are not a team member yet')
+  }
   return unpack(conn('moderator').insert({ team_id: teamId, osm_id: osmId }))
 }
 
 /**
- * Remove a moderator from a team
+ * Remove a moderator from a team.
  * @param {int} teamId - team id
  * @param {int} osmId - osm id
+ * @throws {Error} if the osmId is the only remaining moderator for this team, or if osmId was not a moderator.
  */
 async function removeModerator (teamId, osmId) {
+  const table = 'moderator'
   const conn = await db()
-  return unpack(conn('moderator').where({ team_id: teamId, osm_id: osmId }).del())
+  const moderatorRecord = conn(table).where({ team_id: teamId, osm_id: osmId })
+  const isModerator = parseInt((await unpack(moderatorRecord.count())).count) > 0
```

Comment:
I feel like this is a bit dense, and shouldn't use unpack. We can just use
```
(await moderatorRecord).length
```
since knex will wrap the results into an array.

Reply:
Sounds good- will make this change.